src/data/matrix/basic.lean

```diff
@@ -881,8 +882,8 @@ lemma minor_smul {R : Type*} [semiring R] [add_comm_monoid α] [module R α] (r
   (A : matrix m n α) :
   ((r • A : matrix m n α).minor : (l → m) → (o → n) → matrix l o α) = r • A.minor := rfl
 
-/-- If the minor doesn't repeat elements, then when applied to a diagonal matrix the result is
-diagonal. -/
+/-- Given a `(m × m)` diagonal matrix defined by a map `d : m → α`, if the reindexing arrow `e` is
+  injective, then the resulting matrix is again diagonal. -/
```

Comment:
Let's stay with `map` for functions, I don't want to make the reader feel like we're using category theory here :)
```suggestion
/-- Given a `(m × m)` diagonal matrix defined by a map `d : m → α`, if the reindexing map `e` is
  injective, then the resulting matrix is again diagonal. -/
```

Reply:
Done!